UmatiServerLib/ConvertSimpleValue.cpp

```diff
@@ -44,6 +44,39 @@ copyToVariantFunc ConvertSimpleValue::asVariantFuncArray(std::vector<std::string
   };
 }
 
+copyToVariantFunc ConvertSimpleValue::asVariantFuncArray(std::vector<UmatiServerLib::LocalizedText_t> *variable) {
+  auto pVariable = variable;
+  return [pVariable](UA_Variant *dst) {
+    UA_LocalizedText *tmp = new UA_LocalizedText[pVariable->size()];
+    for (size_t i = 0; i < pVariable->size(); ++i) {
+      tmp[i] = UA_LOCALIZEDTEXT_ALLOC(pVariable->at(i).locale.c_str(), pVariable->at(i).text.c_str());
+    }
+    UA_Variant_setArray(dst, tmp, pVariable->size(), &UA_TYPES[UA_TYPES_LOCALIZEDTEXT]);
+  };
+}
+
+copyToVariantFunc ConvertSimpleValue::asVariantFuncArray(std::vector<std::int32_t> *variable) {
+  auto pVariable = variable;
+  return [pVariable](UA_Variant *dst) {
+    UA_Int32 *tmp = new UA_Int32[pVariable->size()];
+    for (size_t i = 0; i < pVariable->size(); ++i) {
+      tmp[i] = pVariable->at(i);
+    }
+    UA_Variant_setArray(dst, tmp, pVariable->size(), &UA_TYPES[UA_TYPES_INT32]);
+  };
+}
+
+copyToVariantFunc ConvertSimpleValue::asVariantFuncArray(std::vector<std::uint32_t> *variable) {
+  auto pVariable = variable;
+  return [pVariable](UA_Variant *dst) {
+    UA_UInt32 *tmp = new UA_UInt32[pVariable->size()];
+    for (size_t i = 0; i < pVariable->size(); ++i) {
+      tmp[i] = pVariable->at(i);
+    }
+    UA_Variant_setArray(dst, tmp, pVariable->size(), &UA_TYPES[UA_TYPES_UINT32]);
+  };
+}
```

Comment:
Is it possible to catch all primitive datatypes in a single function?

Reply:
It should be possible to make a templated lambda generator for `std::vector<T>*`.